src/app/enterprise.service.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-FileCopyrightText: 2022-present Open Networking Foundation <info@opennetworking.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { TargetsNames } from '../openapi3/top/level/models';
+import { ApiService as TopLevelApiService } from '../openapi3/top/level/services';
+import { tap } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EnterpriseService {
+    enterprises: TargetsNames;
+
+    constructor(private aetherService: TopLevelApiService) {}
+
+    loadTargets() {
+        this.aetherService
+            .targetsTopLevel()
+            .pipe(tap((e) => console.log('Loaded enterprises', e)))
+            .subscribe((enterprises) => (this.enterprises = enterprises));
+    }
+
+    get isSingleton(): boolean {
```

Comment:
Doesn't look like this method is used anywhere, what is it needed for?

Reply:
I was going to use it to hide the `enterprise` column when a user was logged in, with access to only a single enterprise